pkgs/stdenv/darwin/default.nix

```diff
@@ -1362,7 +1434,14 @@ in
           { inherit tools libraries; } // tools // libraries
         );
 
-        inherit (prevStage) binutils binutils-unwrapped;
+        binutils-unwrapped = super.binutils-unwrapped.override {
+          # GNU binutils tries to avoid causing a rebuild of LLVM by using its source from a previous stage,
+          # which results in an evaluation error when binutils is disabled in the bootstrap.
+          # Removing it from `__bootPackages` prevents an evaluation error.
+          stdenv = self.stdenv // {
+            __bootPackages = builtins.removeAttrs self.stdenv.__bootPackages [ "binutils-unwrapped" ];
+          };
```

Comment:
```suggestion
          stdenv = if self.stdenv ? __bootPackages
                   then self.stdenv // {
                          __bootPackages = builtins.removeAttrs self.stdenv.__bootPackages [ "binutils-unwrapped" ];
                       }
                   else self.stdenv;
```

I think it's just because stage 0 doesn't have __bootPackages so this works.

Reply:
I ended up dropping this part because stage 4 doesn’t actually propagate binutils-unwrapped. The point of the throw early is to catch evaluation errors, but the final `allowedRequisites` should catch it if itslips into the stdenv.